sematic/user_settings.py

```diff
@@ -2,61 +2,19 @@
 import enum
 import logging
 import os
-from typing import Dict, Optional
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, List, Optional, Tuple
 
 # Third-party
 import yaml
 
 # Sematic
 from sematic.config_dir import get_config_dir
 
-_SETTINGS_FILE = "settings.yaml"
-
-
 logger = logging.getLogger(__name__)
 
-
-Settings = Dict[str, Dict[str, str]]
-
-
-def _settings_file() -> str:
-    return os.path.join(get_config_dir(), _SETTINGS_FILE)
-
-
-def _load_settings():
-    try:
-        with open(_settings_file(), "r") as f:
-            settings = yaml.load(f, yaml.Loader)
-    except FileNotFoundError:
-        settings = None
-
-    if settings is None:
-        settings = {"default": {}}
-
-    return settings
-
-
-_settings: Optional[Settings] = None
-
-
-def get_all_user_settings() -> Dict[str, str]:
-    """
-    Main API to access stored user settings.
-    """
-    global _settings
-
-    if _settings is None:
-        _settings = _load_settings()
-
-        # Override with env vars
-        for var in SettingsVar.__members__.values():
-            if var.value in os.environ:
-                logger.debug(
-                    "Override {} with {}".format(var.value, os.environ[var.value])
-                )
-                _settings["default"][var.value] = os.environ[var.value]
-
-    return _settings["default"]
+_SETTINGS_FILE = "settings.yaml"
+_DEFAULT_PROFILE = "default"
```

Comment:
Rather than just one default, we could have `default_local` and `default_cloud`. We could pre-populate these with reasonable values. Then we could ditch the undocumented and hacky`switch_env`. One tough thing to figure out there: `sematic start` and `sematic stop` will need to always start/stop using a profile suitable for local execution.

Reply:
Once users switch to cloud, I expect them to stay there, and (almost) never go back to local. Local is really just for onboarding. So it's not so important to make it easy to go back and forth between local and remote.
A single default seems fine to me.

Reply:
@neutralino1's arguments seem valid. However I don't like the current state of `switch_env`. I believe it should be refactored away, maybe as part of #267 (or part of a new, bigger issue), while leveraging this feature.

Reply:
Honestly, if we can't get rid of `switch_env` with this PR, I'm not convinced it's worth the extra code complexity. We are probably the only ones who will be regularly switching between environments in the way supported by this PR and it's easy enough for us to do so by editing our `settings.yaml` files. `switch_env` is the only way users currently have to go between local and remote easily, and it is hacky and undocumented. If we can replace that with this profile mechanism, I'm all for it. Otherwise I'm skeptical.

Reply:
Several observations:

1. Currently it seems the server configs and the user configs are not clearly separated, with some settings which actually configure the server being declared in the `SettingsVar` enum in `user_settings.py`:
    ```
    SEMATIC_AUTHENTICATE
    SEMATIC_AUTHORIZED_EMAIL_DOMAIN
    GOOGLE_OAUTH_CLIENT_ID
    GITHUB_OAUTH_CLIENT_ID
    KUBERNETES_NAMESPACE
    GRAFANA_PANEL_URL
    ```
    Mentioning them in the user's local settings will have no effect. The server gets settings with these names through the user settings API. It's unclear to me at this point if these server settings are sourced from a user settings file or from env variables.

2. The server has a choice between 2 (actually 3, but this is unimportant) configuration templates: "local" and "cloud". These are hydrated exclusively by env variables which are not defined in the user settings enum. The relevant differences between these templates related to this discussion are the address on which the server should bind and the database URL.

    To me, specifying "local" vs "cloud" is shorthand for specifying these web and DB configurations, in lieu of having a server settings file as there exists one for user settings.

3. The server will never actually switch profiles; a server deployment either has a local or a cloud configuration, according to its purpose.

4. Defining a "local" vs "cloud" configuration in the user's local profiles would have no effect, as these profiles control which server backend the Resolver connects to (plus some other resolution-specific settings such as AWS_S3_BUCKET - should this actually be a user setting and not a server one?), and the backend server is already configured, as described above.

5. The current code applications of `switch_env` only seem to be used to ensure the performed operation is pointed at the correct database.

6. This PR is supposed to introduce support for several user local profiles, and an ability to switch between them programatically. The user's action of switching profiles will not have any impact on the server configuration.


In conclusion, I believe there is a mix of concerns here creating confusion. I also believe that the server "local" vs "cloud" sub-optimal configuration scheme should not and cannot be resolved by a profile switching mechanism, but by introducing a separate server settings configuration mechanism backed by persistent storage, similar to the one that exists for user settings, but which does not require profile switching, only configuration generation.

I believe the next steps should be:
1. Create such a configuration mechanism for the server.
2. Move the above-mentioned user settings which actually configure the server from the user settings to the new server settings.
3. Completely remove the usages of `switch_env`.

These steps ^ are unrelated to the current scope of this PR, though.